File: Incomplete/Fizz Buzz Cuckoo Clock.swift
func fizzBuzzCuckooClock(_ time: String) -> String {
  // Do something
  var timeArr = time.split(separator: ":")
  let hour = timeArr[0]
  let minute = timeArr[1]
  print(hour, minute)
  func cuckoo (count: Int) -> String {
    var bird: Array = [String]()
    while (bird.count < count) { 
      bird.append("Cuckoo") 
    }
    return bird.joined(separator: " ");
  }
  
  if (minute == "00") {
    if (hour == "00" ) {
      return cuckoo(count: 12)
    } else if (Int(hour)! <= 12) {
      return cuckoo(count: Int(hour)!)
    } else {
      return cuckoo(count: Int(hour)! - 12)
    }
  } else {
    return "Undefinied"
  }
  
}  

// https://www.codewars.com/kata/fizz-buzz-cuckoo-clock/train/swift
// Server errors

File: JavaScript/7 kyu/Fizz Buzz Cuckoo Clock.js
function fizzBuzzCuckooClock(time) {
  // your code here
  const timeArr = time.split(":");
  const hour = timeArr[0];
  const minute = timeArr[1];
  
  if (minute === "00") {
    let count = hour === "00" ? 12 : 
      parseInt(hour) > 12 ? parseInt(hour) - 12 :
      parseInt(hour);
    let cuckoo = [];
    while (cuckoo.length < count) { cuckoo.push("Cuckoo"); }
    return cuckoo.join(" ");
  } else {
    return parseInt(minute) % 30 === 0 ? "Cuckoo" : 
           parseInt(minute) % 15 === 0 ? "Fizz Buzz" :
           parseInt(minute) % 3 === 0 ? "Fizz" :
           parseInt(minute) % 5 === 0 ? "Buzz" :
           "tick";  
  }
}
